src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3510,14 +3510,27 @@ ExplicitInterfaceSpecifierSyntax tryParseExplicitInterfaceSpecifier()
             }
         }
 
+        private SyntaxToken TryEatCheckedOrHandleUnchecked(ref SyntaxToken operatorKeyword)
+        {
+            if (CurrentToken.Kind == SyntaxKind.UncheckedKeyword)
```

Comment:
@CyrusNajmabadi, is this an example of an error you'd prefer to see done in binding, or is parsing ok for a misplaced token error?

Reply:
Or maybe we can use the existing `ERR_SyntaxError` error?

Reply:
Unfortunately, this needs to be in the parser in the current form.  The reason for that is that the syntax model itself is restrictive here.  e.g. it requires a `SyntaxToken CheckedKeyword`.  As such, we cannot fit the user code into the syntax model, and we have to make a correction at the parsing level, which means having a diagnostic at the parser level.

My overall preference (if we had time/motivation) would be to *relax* the semantic model so that both of checked/unchecked were legal from the model's perspective, and then have it be the binder that checks and states taht 'unchecked' is not actually allowed semantically.

But this is also not a big deal for me (given how rare these will be) so keeping the status quo (and taking this PR) seems totally reasonable to me.

Reply:
I tried relaxing the syntax model to allow either `checked` or `unchecked` keywords, but ran into an unfortunate issue with public APIs...

The `syntax.xml` change updates the API `OperatorMemberCref(SyntaxToken operatorToken, CrefParameterListSyntax? parameters)` to `OperatorMemberCrefSyntax OperatorMemberCref(SyntaxToken checkedKeyword, SyntaxToken operatorToken, CrefParameterListSyntax? parameters)`.

But that conflicts with existing manually defined public API: `OperatorMemberCref(SyntaxToken operatorKeyword, SyntaxToken operatorToken, CrefParameterListSyntax? parameters)`.

@333fred Given that, I think we should keep the parser-level handling.

Reply:
Yeah, we could likely only do this *if* we haven't publicly shipped the 'checked operator' work yet.  if it isn't shipped, then we can revise.  otehrwise, we're locked in.

Reply:
The `checked operator` feature is still in preview. But the public API that is causing a conflict predates that (doesn't involved `checked` at all). So I think we're stuck.

Reply:
@333fred ptal